spec/services/order_invoice_comparator_spec.rb

```diff
@@ -115,19 +115,217 @@
       end
     end
 
-    context "a non-relevant associated model is updated" do
-      let(:distributor){ order.distributor }
-      it "returns false" do
-        distributor.update!(name: 'THIS IS A NEW NAME', abn: 'This is a new ABN')
-        expect(subject).to be false
+    context "changes on an associated order object" do
+      describe "detecting relevant associated object changes" do
+        context "adjustment changes" do
+          it "creating a new adjustment returns true" do
+            create(:adjustment, order_id: order.id)
+            order.reload
```

Comment:
Wait, I just figured it out, I think this would work (also removing the need for `order.reload`)

```suggestion
            order.adjustments << create(:adjustment)
```

Reply:
Nice catch, thanks @dacook . 
I noticed though, that although adjustments array was being populated with a new adjustment, it was being created without an `order_id`; this seemed to make the tests fail.

By adding `order:` to it, made the tests pass again, like so:
`order.adjustments << create(:adjustment, order:)`